test/mailinSpec.js

```diff
@@ -16,15 +16,34 @@ chai.use(chaiAsPromised);
 var should = null;
 should = chai.Should();
 
+var server = express(),
+    conn;
+var doing = 0;
+
 before(function (done) {
-    mailin.start({
-        // verbose: true,
-        smtpOptions: {
-            secure: false
-        }
-    }, function (err) {
+    /* Make an http server to receive the webhook. */
+    server.head('/webhook', function (req, res) {
+        res.send(200);
+        doing--;
```

Comment:
This one actually set `doing` to `-1`, for it is called before the test itself can set it doing to the required value.